registry/proxy/proxyblobstore.go

```diff
@@ -67,6 +69,29 @@ func (pbs *proxyBlobStore) copyContent(ctx context.Context, dgst digest.Digest,
 	return desc, nil
 }
 
+func (pbs *proxyBlobStore) doResponse(reqMethod string, ctx context.Context, dgst digest.Digest, writer io.Writer) error {
+	switch reqMethod {
+	case http.MethodHead:
+		desc, err := pbs.remoteStore.Stat(ctx, dgst)
+		if err != nil {
+			return err
+		}
+
+		if w, ok := writer.(http.ResponseWriter); ok {
+			setResponseHeaders(w, desc.Size, desc.MediaType, dgst)
+		}
+		return nil
+	case http.MethodGet:
+		if _, err := pbs.copyContent(ctx, dgst, writer); err != nil {
+			return err
+		}
+		return nil
+	default:
+		return errors.New(fmt.Sprintf("BlobStore: unsupported request method[%s]\n", reqMethod))
```

Comment:
Please use `errors.Errorf` from `github.com/pkg/errors` and error strings should not start with a capital letter